controllers/components/codeflare/codeflare_support.go

```diff
@@ -10,6 +10,11 @@ import (
 
 const (
 	ComponentName = componentApi.CodeFlareComponentName
+
+	// LegacyComponentName is the name of the component that is assigned to deployments
+	// via Kustomize. Since a deployment selector is immutable, we can't upgrade existing
+	// deployment to the new component name, so keep it around till we figure out a solution.
+	LegacyComponentName = "codeflare"
```

Comment:
@lburgazzoli Question, shouldn't we be making the LegacyComponentName be different from the component name? Just to identity which one we need to figure out a solution for?

Reply:
I don't think so. `LegacyComponentName`  is not meant to last forever (hopefully) but we can't just rename it because we have some legacy usage here and there we can't just removed.